OrderService/OrderService.API/Controllers/OrderController.cs

```diff
@@ -0,0 +1,150 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using OrderService.Application.DTO;
+using OrderService.Application.UseCases.OrderUseCases.CancelOrderUseCase;
+using OrderService.Application.UseCases.OrderUseCases.CompleteOrderUseCase;
+using OrderService.Application.UseCases.OrderUseCases.ConfirmOrderUseCase;
+using OrderService.Application.UseCases.OrderUseCases.CreateOrderUseCase;
+using OrderService.Application.UseCases.OrderUseCases.GetAllOrdersUseCase;
+using OrderService.Application.UseCases.OrderUseCases.GetOrderByIdUseCase;
+using OrderService.Application.UseCases.OrderUseCases.GetUserOrdersUseCase;
+using OrderService.Application.Models;
+using OrderService.Application.UseCases.PaymentUseCases.PayOrderUseCase;
+using AutoMapper;
+
+namespace OrderService.API.Controllers;
+
+[Route("api/orders")]
+[ApiController]
+public class OrderController : ControllerBase
+{
+    private readonly IMediator _mediator;
+    private readonly IMapper _mapper;
+
+    public OrderController(IMediator mediator, IMapper mapper)
+    {
+        _mediator = mediator;
+        _mapper = mapper;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateOrder(CancellationToken cancellationToken)
+    {
+        var userId = HttpContext.User.FindFirst("Id")?.Value;
+
+        await _mediator.Send(new CreateOrderRequest(userId), cancellationToken);
+
+        return NoContent();
+    }
+
+    [HttpGet]
+    [Authorize]
+    public async Task<ActionResult<PaginatedListModel<OrderDTO>>> GetOrders(CancellationToken cancellationToken,
+        [FromQuery] PaginationDTO pagination)
+    {
+        var request = _mapper.Map<GetUserOrdersRequest>(pagination);
+        request.UserId = HttpContext.User.FindFirst("Id")?.Value;
+
+        var data = await _mediator.Send(request, cancellationToken);
+
+        return Ok(data);
+    }
+
+    [HttpGet("all")]
+    [Authorize(Policy = "admin")]
+    public async Task<ActionResult<PaginatedListModel<OrderDTO>>> GetAllOrders(CancellationToken cancellationToken,
+        [FromQuery] PaginationDTO pagination)
+    {
+        var request = _mapper.Map<GetAllOrdersRequest>(pagination);
+
+        var data = await _mediator.Send(request, cancellationToken);
+
+        return Ok(data);
+    }
+
+    [HttpGet("user/{userId:regex(^[[a-fA-F0-9]]{{24}}$)}")]
```

Comment:
please use plural nouns in routes
and it's better to name it like api/users/{userId}/orders